sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.ChangeFeedProcessor;
+import com.azure.cosmos.util.Beta;
+
+import java.time.Instant;
+
+/**
+ * Specifies the {@link ChangeFeedProcessor} state for a particular lease/worker.
+ */
+@Beta(Beta.SinceVersion.V4_2_0)
+public class ChangeFeedProcessorState {
+    private String id;
+    private String leaseToken;
+    private String hostName;
+    private Instant lastUpdate;
+    private String continuationToken;
+    private Instant continuationTokenTimestamp;
+    private int estimatedLag;
+
+    /**
+     * Gets the ID of the lease item representing the persistent state of a change feed processor worker.
+     *
+     * @return the ID of the lease item.
+     */
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Sets the ID of the lease item representing the persistent state of a chenge feed processor worker.
+     *
+     * @param id a unique string.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Gets the token of the lease item representing the persistent state of a change feed processor worker.
+     * <p>
+     * A lease token is a unique identifier representing a specific scope that a CFP worker will operate on.
+     *
+     * @return a string token uniquely representing the scope for one worker unit.
+     */
+    public String getLeaseToken() {
```

Comment:
Capturing the comment from above...

This would need to be of type FeedRange soon. https://github.com/FabianMeiswinkel/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/FeedRange.java

I think we have 3 Options:

Keep it String here, to avoid breaking change use the FeedRange.toSjon as the value and expect customers to do the transformations upstream
Make a breaking change to this preview contract when I merge the FeedRange changes - would probably be good to set expectations with Walmart that we expect some breaking changes here
Add FeedRange as an additional field - make Id the FeedRange.toJson() - this way it doesn't break and the Id might not be very useful anymore but also doesn't really cause harm
I think 3) is an acceptable solution - le me know whether you agree/disagree

Reply:
The lease token can be any string representing one scope for a CFP worker to operate on, including a JSON string if we want to. It will be TBD how we store it in the lease document/item to make it persistent but from the user's perspective it should not matter much since FeedRange already has a constructor that takes a string payload.
We can also consider adding a "version" property to distinguish between different implementations if that helps.